internal/topo/connection/client_factory.go

```diff
@@ -0,0 +1,78 @@
+package connection
+
+import (
+	"fmt"
+	"github.com/lf-edge/ekuiper/internal/conf"
+)
+
+type ClientFactoryFunc func(super Client) ClientCreator
+
+type ClientCreator interface {
+	CfgValidate(props map[string]interface{}) error
+	GetClient() (interface{}, error)
+}
+
+type Client struct {
```

Comment:
This name seems misleading, because it is just some configuration information instead of a client.